unity-client/Assets/Scripts/MainScripts/DCL/Controllers/PointerEventsController.cs

```diff
@@ -35,7 +35,10 @@ public void Initialize()
             InputController_Legacy.i.AddListener(WebInterface.ACTION_BUTTON.PRIMARY, OnButtonEvent);
             InputController_Legacy.i.AddListener(WebInterface.ACTION_BUTTON.SECONDARY, OnButtonEvent);
 
-            hoverController = Environment.i.interactionHoverCanvasController;
+            hoverController = InteractionHoverCanvasController.i;
+
+            OnPointerHoverStarts += CursorController.i.SetHoverCursor;
+            OnPointerHoverEnds += CursorController.i.SetNormalCursor;
```

Comment:
The rest of the `InteractionHoverCanvasController` dependency inversion referred on the other comment.